### DISCUSSION TITLE
Does run_maintenance() require all partitions for a parent table to be managed by pg_partman?

### DISCUSSION THREAD

**Discussion by TonyJulin** [pgpartman/pg_partman#675] [Q&A] [answered]
Hello again. 
In a previous post, I mentioned our partition intervals (and current, manually configured partition tables) to have the format of _table_partition_archive_, _table_partition_2018_, and _table_partition_2022_january_. I've decided to go the route (for now) of using partman for the monthly table partitioning, and then we can manually configure the yearly and archive partitions. However, the snag I'm running into is that when `run_maintenance()` is called on the parent table, it's expecting *all* partitions to be in the proper, pg_partman format. So the '_archive' partition is causing the failure: 
_Error executing query:
SQL Error [P0001]: ERROR: invalid value "arch" for "YYYY"_

Looking at the docs, I don't see a way of excluding certain partitions from being maintained. Is there something I'm missing or some workaround? Thanks!

**Comment by keithf4** [comment]
Did you find out what version of pg_partman you were running? Partman 4.x relies on parsing the child table names, so custom child table names are not going to work in that version.
For version 5.x, it's looking at the actual partition constraints. I'm no 100% sure if it will support differently named children at this point, but if it's not working, that's something I could look into. However, the partition constraints would definitely have to match the overall partitioning type for the non-standard child tables (Ex they'd all have to be time-based).

**Reply by TonyJulin** [reply]
I confirmed that the version being used is 4.7.3 for our Aurora version (AWS docs have a discrepancy, go figure) and there are no plans as of yet for AWS to update their Aurora engines to 5.0.0+. With what you said, my understanding then is that there is no way to keep our "_archive" partition table then, even if we only want to pg_partman to maintain monthly partition tables?
For reference, our '_archive' partition is for entries that are > 5 years old going back to x years. In that case, for us to be able to use pg_partman for that table we'd have to then break it out into tables like "_archive_2015", "_archive_2016", etc.

**Reply by keithf4** [reply, marked as answer]
Yes for version 4.x you must have the partition interval in the partition name in a format that partman expects.

**Reply by TonyJulin** [reply]
Thank you! I'll see about splitting up that partition into a format acceptable by partman.
